contract/src/lib.rs

```diff
@@ -22,60 +73,115 @@ impl Default for Contract {
 pub enum StorageKeys {
     PurchasesPerAccount,
     PurchasesPerAccountInner { account_hash: CryptoHash },
+    PaidContentPerContext,
+    PaidContentPerContextInner { account_hash: CryptoHash },
+    PaidContents,
+}
+
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
 }
 
 #[near_bindgen]
 impl Contract {
-    #[payable]
-    pub fn buy(&mut self, content_id: ContentId) -> Promise {
-        let account_id = env::predecessor_account_id();
-        let mut content_purchases =
-            self.purchases_per_account
-                .get(&account_id)
-                .unwrap_or_else(|| {
-                    UnorderedSet::new(StorageKeys::PurchasesPerAccountInner {
-                        account_hash: env::sha256_array(account_id.as_bytes()),
-                    })
-                });
+    pub fn add_paid_content(&mut self, link: String, cost: String, context_id: String) -> String {
+        let author = env::predecessor_account_id();
+        let id = calculate_hash(&(context_id.clone() + &link)).to_string();
```

Comment:
Values `{ context_id: '123', link: '456' }` and `{ context_id: '1', link: '23456' }` will produce the same hash.

Reply:
fixed